packages/@aws-cdk/aws-logs/lib/data-protection-policy.ts

```diff
@@ -0,0 +1,180 @@
+
+  /**
+   * Name of the data protection policy
+   *
+   * @default - 'data-protection-policy-cdk'
+   */
+  public readonly name: string;
+
+  /**
+   * Description of the data protection policy
+   *
+   * @default - 'cdk generated data protection policy'
+   */
+  public readonly description: string;
+
+  /**
+   * Version of the data protection policy
+   *
+   * @default - '2021-06-01'
+   */
+  public readonly version: string;
+
+  /**
+   * Statements within the data protection policy. Must contain one Audit and one Redact statement
+   *
+   * @default - one AuditStatement and one RedactStatement
+   */
+  public statement: readonly [AuditStatement, RedactStatement];
+
+  constructor(props: DataProtectionPolicyProps) {
+    this.name = props.name || 'data-protection-policy-cdk';
+    this.description = props.description || 'cdk generated data protection policy';
+    this.version = '2021-06-01';
+
+    var findingsDestination: FindingsDestination = {};
+
+    if (props.logGroupNameAuditDestination) {
+      findingsDestination.cloudWatchLogs = {
+        logGroup: props.logGroupNameAuditDestination,
+      };
+    }
+
+    if (props.s3BucketNameAuditDestination) {
+      findingsDestination.s3 = {
+        bucket: props.s3BucketNameAuditDestination,
+      };
+    }
+
+    if (props.deliveryStreamNameAuditDestination) {
+      findingsDestination.firehose = {
+        deliveryStream: props.deliveryStreamNameAuditDestination,
+      };
+    }
+
+    var identifierArns: string[] = [];
+    for (let identifier of props.identifiers) {
+      identifierArns.push('arn:aws:dataprotection::aws:data-identifier/' + identifier);
+    };
+
+    var auditStatement: AuditStatement = {
+      sid: 'audit-statement-cdk',
+      dataIdentifier: identifierArns,
+      operation: {
+        audit: {
+          findingsDestination: findingsDestination,
+        },
+      },
+    };
+
+    var redactStatement: RedactStatement = {
+      sid: 'redact-statement-cdk',
+      dataIdentifier: identifierArns,
+      operation: {
+        deidentify: {
+          maskConfig: {},
+        },
+      },
+    };
+
+    this.statement = [auditStatement, redactStatement];
+  }
+
+}
+
+type AuditStatement = {
+  sid: string;
+  dataIdentifier: string[];
+  operation: AuditOperation;
+}
+
+type AuditOperation = {
+  audit: FindingsDestinations;
+}
+
+type FindingsDestinations = {
+  findingsDestination: FindingsDestination;
+}
+
+type FindingsDestination = {
+  cloudWatchLogs?: CloudWatchLogsDestination;
+  firehose?: FirehoseDestination;
+  s3?: S3Destination;
+}
+
+type CloudWatchLogsDestination = {
+  logGroup: string;
+}
+
+type FirehoseDestination = {
+  deliveryStream: string;
+}
+
+type S3Destination = {
+  bucket: string;
+}
+
+type RedactStatement = {
+  sid: string;
+  dataIdentifier: string[];
+  operation: DeidentifyOperation;
+}
+
+type DeidentifyOperation = {
+  deidentify: MaskConfig;
+}
+
+type MaskConfig = {
+  maskConfig: {}
+}
```

Comment:
Style suggestion: A lot of this seems unnecessary since you're not doing much with these values. I don't really see a reason to not just plug in the values directly with something like:
```
    this.statement = {
      sid: 'audit-statement-cdk',
      dataIdentifier: props.identifiers,
      operation: {
        audit: {
          findingsDestination: {
            Object.fromEntries(props.auditDestinations),
          },
        },
      },
      {
        sid: 'redact-statement-cdk',
        dataIdentifier: props.dataIdentifiers,
        operation: {
          deidetify: {
            maskConfig: {},
          },
        },
      },
    };
```

Reply:
I've removed some of these types in favor of your style.